tests/unit/prettier-version.test.js

```diff
@@ -1,26 +1,19 @@
-const path = require('path');
-
-jest.mock('prettier', () => {
-  // Require the original module to not be mocked...
-  const originalModule = jest.requireActual('prettier');
+const prettier = require('prettier');
+const proxyquire = require('proxyquire');
 
-  return {
-    ...originalModule,
-    version: '2.2.1'
-  };
+const plugin = proxyquire('../../src', {
+  prettier: { ...prettier, version: '2.2.1', '@global': true }
 });
 
-const prettier = require('prettier');
-
 test('should throw if the installed version of prettier is less than v2.3.0', () => {
   const data = 'contract CheckPrettierVersion {}';
 
   const options = {
-    plugins: [path.join(__dirname, '../..')],
+    plugins: [plugin],
     parser: 'solidity-parse'
   };
 
-  expect(() => {
-    prettier.format(data, options);
-  }).toThrow('>=2.3.0');
+  expect(async () => {
```

Comment:
Does this work? Shouldn't the test function be async and use `await expect` here too?

Reply:
I tried a few combos when I was cleaning this test. it looks like `expect().reject.toThrow()` expects a promise as the `expect` argument as the test would fail otherwise.

Reply:
Sorry, I meant that you should also `await` the result of `expect`, at least that's how it's done in other tests, and how a similar matcher in chai works. Otherwise all the code in the test would be sync, but you are asserting the result of an async function, which can't really be done. Does that make sense?

Reply:
I addressed this one